dbt/adapters/databricks/connections.py

```diff
@@ -207,17 +217,52 @@ def close(self) -> None:
         except Error as exc:
             logger.warning(CursorCloseError(self._cursor, exc))
 
-    def fetchall(self) -> Sequence[tuple]:
+    def fetchall(self) -> Sequence[Row]:
         return self._cursor.fetchall()
 
-    def fetchone(self) -> Optional[tuple]:
+    def query_impact(self) -> DatabricksQueryImpact:
```

Comment:
little nit on the name maybe you can call it `get_query_statistics` - I'm not a native English speaker, but in other query engines the term statistics is used for this type of information.
If you like this name more, you should consider to replace `impact` with `statistics` everywhere for consistency.